core/corehttp/gateway_handler.go

```diff
@@ -499,6 +499,8 @@ func (i *gatewayHandler) putHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	i.node.Provider.Provide(newcid)
+
```

Comment:
@eingenito all those paths are covered by the provider calls in `CoreAPI`. The reason you see some provides in `gateway_handler` is because, at the time of writing this code, there were some aspects of gateway_handler that were not covered by coreapi.